src/main/java/org/spongepowered/common/data/provider/DataProviderRegistry.java

```diff
@@ -100,10 +101,10 @@ private static boolean filterHolderType(DataProvider<?,?> provider, Class<?> hol
     }
 
     private DataProviderLookup loadProviderLookup(Class<?> holderType) {
-        return new DataProviderLookup(this.dataProviders.keySet().stream()
-                .map(key -> this.getProvider(key, holderType))
-                .filter(provider -> !(provider instanceof EmptyDataProvider))
-                .collect(Collectors.toMap(DataProvider::getKey, Function.identity())));
+        Stream<DataProvider> stream = this.dataProviders.keySet().stream()
+                .map(key -> this.getProvider((Key) key, holderType))
+                .filter(provider -> !(provider instanceof EmptyDataProvider));
+        return new DataProviderLookup(stream.collect(Collectors.toMap(p -> (Key<?>) p.getKey(), p -> (DataProvider<?, ?>) p)));
```

Comment:
This is really janky to read. Either collect on the stream above or declare a new variable to pass in.

Reply:
Collection on the stream caused incompatible types compile errors